internal/security/config/command/proxy/adduser/command.go

```diff
@@ -0,0 +1,340 @@
+	if err != nil {
+		_ = resp.Body.Close()
+		return fmt.Errorf("Failed to send new consumer request %s: %w", c.username, err)
+	}
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	_ = resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+	case http.StatusCreated:
+		c.loggingClient.Info(fmt.Sprintf("created consumer (user) '%s'", c.username))
+	case http.StatusConflict:
+		c.loggingClient.Info(fmt.Sprintf("consumer '%s' already created", c.username))
+	default:
+		c.loggingClient.Error(fmt.Sprintf("%s", responseBody))
+		return fmt.Errorf("Create consumer request failed with code: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func (c *cmd) addUserToGroup() error {
+	// Associate the consumer with a group
+	// https://docs.konghq.com/hub/kong-inc/acl/#associating-consumers
+
+	form := url.Values{
+		"group": []string{c.group},
+	}
+	kongURL := strings.Join([]string{c.configuration.KongURL.GetProxyBaseURL(), "consumers", c.username, "acls"}, "/")
+	c.loggingClient.Info(fmt.Sprintf("Associating consumer to acl using endpoint %s", kongURL))
+
+	formVal := form.Encode()
+	req, err := http.NewRequest(http.MethodPost, kongURL, strings.NewReader(formVal))
+	if err != nil {
+		return fmt.Errorf("Failed to build request to associate consumer %s to group %s: %w", c.username, c.group, err)
+	}
+	req.Header.Add(clients.ContentType, urlEncodedForm)
+	resp, err := c.client.Do(req)
+	if err != nil {
+		_ = resp.Body.Close()
+		return fmt.Errorf("Failed to subnit request to associate consumer %s to group %s: %w", c.username, c.group, err)
+	}
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	_ = resp.Body.Close()
+	if err != nil {
+		return err
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+	case http.StatusCreated:
+		c.loggingClient.Info(fmt.Sprintf("associated consumer %s to group %s", c.username, c.group))
+	case http.StatusConflict:
+		c.loggingClient.Info(fmt.Sprintf("consumer %s already associated to group %s", c.username, c.group))
+	default:
+		c.loggingClient.Error(fmt.Sprintf("%s", responseBody))
+		return fmt.Errorf("Failed to associate consumer to group with status: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func (c *cmd) ExecuteAddJwt() (int, error) {
+	publicKey, err := ioutil.ReadFile(c.publicKeyPath)
+	if err != nil {
+		return interfaces.StatusCodeExitWithError, fmt.Errorf("Failed to read public key from file %s: %w", c.publicKeyPath, err)
+	}
+
+	if err := c.createConsumer(); err != nil {
+		return interfaces.StatusCodeExitWithError, err
+	}
+	if err := c.addUserToGroup(); err != nil {
+		return interfaces.StatusCodeExitWithError, err
+	}
+
+	// Associate a JWT credential with the consumer
+	// https://docs.konghq.com/hub/kong-inc/jwt/#create-a-jwt-credential
+
+	form := url.Values{
+		"algorithm":      []string{c.algorithm},
+		"rsa_public_key": []string{string(publicKey)},
+		"secret":         []string{"required-bot-not-used-see-documentation"},
+	}
+
+	if len(c.jwtID) > 0 {
```

Comment:
Is OK for this to be empty?

Reply:
Yes.  If not passed in Kong generates a random one and returns it in the result.